### PR TITLE
test(console): cover janitor lifecycle edge cases

### PR DESCRIPTION
**Pull request by march26bot**
$## Summary
- cover the exact-expiry ticket boundary in the console store
- verify the background janitor removes expired tickets automatically
- verify Close stops the janitor so outstanding tickets are not pruned after shutdown

## Testing
- Not run (Go toolchain unavailable in this runtime)

### CODE REVIEW

File: internal/console/store_test.go [hunk 1]
```diff
@@ -103,6 +136,48 @@ func TestJanitorRemovesExpiredTickets(t *testing.T) {
...
+	s := NewStoreWithOptions(20*time.Millisecond, 10*time.Millisecond)
+	defer s.Close()
+	s.now = func() time.Time { return now }
+
+	token, _, err := s.IssueTicket("vm-1", "api-key-1")
+	if err != nil {
+		t.Fatalf("IssueTicket returned error: %v", err)
+	}
+
+	now = now.Add(time.Minute)
```
Comment (chinmay28) [inline internal/console/store_test.go:150]: This test has a data race. `NewStoreWithOptions` starts the janitor goroutine immediately (`store.go:60` → `startJanitor`), and with a 10ms interval it ticks during the test. The janitor reads `s.now` and calls it (`store.go:128` `s.deleteExpired(s.now())`), while this test concurrently **writes** `s.now` (line 143) and mutates the captured `now` variable (line 150) — both unsynchronized.

`go test -race` (which `make test` uses) reports it:

```
WARNING: DATA RACE
Read at ... by goroutine 8:
  ...(*Store).startJanitor.func1()  store.go:128
Previous write at ... by goroutine 7:
  ...TestJanitorRemovesExpiredTicketsAutomatically()  store_test.go:143
...
Read at ... store_test.go:143  (closure reading `now`)
Previous write at ... store_test.go:150  (`now = now.Add(...)`)
--- FAIL: TestJanitorRemovesExpiredTicketsAutomatically
```

The existing janitor test avoids this by using a `time.Hour` interval so the goroutine never ticks. For an *automatic*-removal test you need a real tick, so the clock has to be concurrency-safe. Suggestion: back the mock clock with an atomic and never reassign `s.now` after construction, e.g.

```go
var nowNanos atomic.Int64
nowNanos.Store(time.Date(2026, 5, 6, 0, 0, 0, 0, time.UTC).UnixNano())
s := NewStoreWithOptions(20*time.Millisecond, 10*time.Millisecond)
defer s.Close()
s.now = func() time.Time { return time.Unix(0, nowNanos.Load()) }
// ... IssueTicket ...
nowNanos.Add(int64(time.Minute))   // safe concurrent advance
```

(There's still a brief window where the janitor may run with the default `time.Now` between construction and the `s.now =` assignment, but since it only deletes expired tickets that's harmless here. `TestCloseStopsJanitor` below shares the same construct-then-assign pattern — it passed for me but is latently racy for the same reason; folding both onto an atomic clock fixes them together.)

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by chinmay28 — CHANGES_REQUESTED** [review]
Good edge cases to cover — the exact-expiry boundary and the automatic janitor sweep are both worth locking in. One blocker though:

- ✅ `TestConsumeTicket_ExpiryBoundaryIsExpired` is correct and matches the `!now.Before(t.expires)` semantics (at exactly `expiresAt` the ticket is expired). Nice.
- ❌ `TestJanitorRemovesExpiredTicketsAutomatically` has a **data race** that fails `go test -race` (i.e. `make test`). The 10ms-interval janitor goroutine reads `s.now`/`now` while the test writes them. See the inline comment for the race report and a fix using an atomic-backed clock.
- ⚠️ `TestCloseStopsJanitor` passed in my runs but shares the same construct-then-reassign-`s.now` pattern; the atomic-clock fix covers it too.

I ran `CGO_ENABLED=1 go test -race ./internal/console/ -count=5` to confirm — the race trips reliably. Once the auto-removal test uses a concurrency-safe clock this is good to go; happy to re-review.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: internal/console/store_test.go [hunk 2]
```diff
@@ -103,6 +137,52 @@ func TestJanitorRemovesExpiredTickets(t *testing.T) {
 	}
 }
 
+func TestJanitorRemovesExpiredTicketsAutomatically(t *testing.T) {
+	var nowNanos atomic.Int64
+	nowNanos.Store(time.Date(2026, 5, 6, 0, 0, 0, 0, time.UTC).UnixNano())
+
+	s := NewStoreWithOptions(20*time.Millisecond, 10*time.Millisecond)
+	defer s.Close()
+	s.now = func() time.Time { return time.Unix(0, nowNanos.Load()).UTC() }
```
Comment (chinmay28) [inline internal/console/store_test.go]: The atomic `nowNanos` doesn't address the race that the earlier review flagged. `NewStoreWithOptions` starts the janitor goroutine at `store.go:60` **before** this line reassigns the `s.now` field. The race is on the function-value field itself, not the int it closes over:

```
Read at ... by goroutine 8:
  (*Store).startJanitor.func1()  store.go:128   // janitor reads s.now()
Previous write at ... by goroutine 7:
  TestJanitorRemovesExpiredTicketsAutomatically()  store_test.go:146   // test writes s.now = ...
```

Making the clock atomic protects the *value* read, but `s.now = func()…` is still an unsynchronized write to the field while the janitor reads it. `go test -race ./internal/console/ -count=5` trips reliably on this branch (and `TestCloseStopsJanitor` does the same reassignment at line 167).

Two clean options:

1. **Test-only (preferred for this PR):** drop the `s.now` injection entirely and lean on the real clock — the test already uses a 20ms TTL / 10ms janitor interval, so a real-time `waitForCondition` will observe the sweep without ever touching `s.now`:
   ```go
   s := NewStoreWithOptions(20*time.Millisecond, 10*time.Millisecond)
   defer s.Close()
   token, _, err := s.IssueTicket("vm-1", "api-key-1")
   // ... err check ...
   waitForCondition(t, time.Second, func() bool {
       s.mu.RLock(); defer s.mu.RUnlock()
       _, ok := s.tickets[token]; return !ok
   })
   ```
   For `TestCloseStopsJanitor`, issue → `Close()` → real `time.Sleep` past TTL → assert still present.

2. **Production change:** add a clock option to the constructor so `now` is set *before* `startJanitor()`, keeping the field immutable after construction.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by chinmay28 — CHANGES_REQUESTED** [review]
The two new scenarios — exact-expiry boundary and the background janitor sweep — are the right things to lock in, and `TestConsumeTicket_ExpiryBoundaryIsExpired` is correct.

But the `4bebcc2` "use atomic clock" commit doesn't actually fix the data race the previous review flagged. I ran `CGO_ENABLED=1 go test -race ./internal/console/ -count=5` on this branch and it still fails:

```
--- FAIL: TestJanitorRemovesExpiredTicketsAutomatically (0.02s)
    testing.go:1490: race detected during execution of test
```

The race isn't on the clock *value* (that's now atomic) — it's the reassignment of the `s.now` **function field** after `NewStoreWithOptions` has already started the janitor goroutine, which reads `s.now()` on its ticker. See the inline comment for the race report and two fixes (preferred: drop the `s.now` injection and use the real clock, since the test already uses 20ms/10ms timings).

Since this is a test-only PR meant to harden `make test`, it can't itself break `go test -race`. Happy to re-review once the race is gone.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: internal/console/store_test.go (hunk 1 above)
Comment (march26bot) [reply internal/console/store_test.go:150]: Fixed the same root cause in `TestCloseStopsJanitor` by dropping the post-construction `s.now` reassignment there too. The test now issues a short-lived ticket, calls `Close()`, sleeps past TTL/janitor interval, and asserts the ticket is still present.

File: internal/console/store_test.go (hunk 2 above)
Comment (march26bot) [reply internal/console/store_test.go]: Switched both janitor tests to use the real clock/ticker path instead of reassigning `s.now` after `NewStoreWithOptions()` starts the goroutine. That removes the unsynchronized `s.now` field write entirely while still covering automatic expiry and post-Close behavior.

File: internal/console/store_test.go (hunk 1 above)
Comment (march26bot) [reply internal/console/store_test.go:150]: Fixed the same root cause in `TestCloseStopsJanitor` by dropping the post-construction `s.now` reassignment there too. The test now issues a short-lived ticket, calls `Close()`, sleeps past TTL/janitor interval, and asserts the ticket is still present.

Comment (march26bot) [reply internal/console/store_test.go:150]: Addressed in commit 60dce5d — I removed the post-construction `s.now` injection and switched the test to use the real clock with `waitForCondition`, so the janitor can tick naturally without racing the test.

File: internal/console/store_test.go (hunk 2 above)
Comment (march26bot) [reply internal/console/store_test.go]: Addressed in commit 60dce5d — both janitor tests now avoid reassigning `s.now` after `NewStoreWithOptions`, which removes the function-field race entirely. `TestJanitorRemovesExpiredTicketsAutomatically` waits on real time; `TestCloseStopsJanitor` closes the store before sleeping past the TTL and then asserts the ticket is still present.

**Review by chinmay28 — APPROVED** [review]
Approving — the data race flagged in the earlier reviews is resolved in `60dce5d`.

Verified locally (`internal/console` has no libvirt dependency, so I could run it directly):

- `CGO_ENABLED=1 go test -race ./internal/console/ -count=5` → clean.
- `go test -race -count=10 -run 'Janitor|Close|ExpiryBoundary'` → all four cases pass 10/10, zero race reports.

The fix is the right one: both janitor tests now lean on the real clock + ticker (20ms TTL / 10ms interval) and a polling `waitForCondition` instead of reassigning the `s.now` function field after `NewStoreWithOptions` has already started the goroutine, which is what tripped `-race`. `TestConsumeTicket_ExpiryBoundaryIsExpired` keeps the `s.now` injection but is safe because its janitor interval is `time.Hour` (the goroutine never ticks during the test).

Correctness checks against `store.go`:
- Exact-expiry boundary: `ConsumeTicket` uses `!now.Before(t.expires)`, so consuming at exactly `expiresAt` is expired — the assertion matches.
- `TestCloseStopsJanitor`: ticket TTL (20ms) outlives the immediate `Close()`, and even a stray pre-Close tick at 10ms wouldn't delete a not-yet-expired ticket, so the "still present after Close" assertion is sound (not flaky).

Good hardening of `make test`. No production code touched.

---
_Generated by [Claude Code](https://claude.ai/code)_
